core/domain/skill_domain.py

```diff
@@ -566,7 +570,7 @@ def validate(self):
             difficulties_list.append(rubric.difficulty)
             rubric.validate()
 
-        if len(difficulties_list) != 3:
+        if len(difficulties_list) != 3 and not new_skill:
```

Comment:
In order to force canonicalization, we should also specify the order of the elements in the list (until such time when we decide that e.g. there are a variable number of difficulty levels).

Reply:
In the frontend, when the skill is being created, I am enforcing the order, though should there be a validation against it? Since, whatever order is present, the rubrics can be keyed by the difficulty value itself to get the right one at any point.

Reply:
Yes, please validate the order for now -- we can relax it later.

Reply:
Done